src/session/session_api.c

```diff
@@ -1583,55 +1584,94 @@ __wt_session_range_truncate(
     trunc_info->stop = stop;
     trunc_info->orig_start_key = orig_start_key;
     trunc_info->orig_stop_key = orig_stop_key;
-    if (uri != NULL)
-        trunc_info->uri = uri;
-    else if (start != NULL)
-        trunc_info->uri = start->internal_uri;
-    else {
-        WT_ASSERT(session, stop != NULL);
-        trunc_info->uri = stop->internal_uri;
-    }
+    trunc_info->uri = actual_uri;
+
+    /*
+     * We would like to use bounded cursors if possible, so check whether they are supported.
```

Comment:
Can you comment on the conditions that make them not supported, or supported?

Reply:
Would something like this work?

```
    /*
     * We would like to use bounded cursors if possible, so check whether they are supported. We
     * don't currently support bounded cursors for fixed column stores. For now, also fall back to
     * not using bounded cursors for complex types, such as tables with column groups and indexes,
     * which may or may not support bounded cursors (which is not easy to check at this abstraction
     * level).
     */
```

The condition is a little lazy as it doesn't check whether a complex table with column groups and indexes supports. Unless there isn't a quick way to do this, I think it's okay to keep the check as it is given how little these features are used. Let me know if you think differently.

Reply:
Where in the code do you exclude tables with column groups?

Reply:
They have `CUR2BT(start) == NULL`.